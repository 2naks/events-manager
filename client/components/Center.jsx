import React, { Component } from 'react';
import DeleteModal from './DeleteModal.jsx';

export default class Center extends Component {

  constructor(props) {
    super(props);
  }
  
  render() {
    return (
        <div className="col-md-4 event">
          <div className="card">
            <div className="card-block">
              <h5 className="card-title"><b>{this.props.centerDetails.name}</b></h5>
              <p className="card-text h6" >{this.props.centerDetails.description.substr(0, 50) + '...'}</p>
              <p className="card-text h6"><b>Location</b>:{this.props.centerDetails.location.substr(0, 25) + '...'}</p>
              <p className="card-text h6"><b>Capacity</b>: {this.props.centerDetails.capacity} seats</p>
              <p className="card-text h6"><b>Price</b>: &#8358;{this.props.centerDetails.price}</p>
              <p className="text-center"><a href="#" className="btn btn-sm btn-primary" onClick={ () => this.props.show('centerDetails', this.props.centerDetails.id) } >
              <i className="fa fa-info-circle fa-lg"></i> View Center</a>
              <a href="#" className="btn btn-sm btn-success" onClick={ () => this.props.show('modifyCenter', this.props.centerDetails.id) } >
              <i className="fa fa-edit fa-lg"></i> Modify</a></p>
            </div>
          </div>
          <DeleteModal objectId={1}  />
        </div>
    );
  }
}
