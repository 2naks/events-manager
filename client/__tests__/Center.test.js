import Center from '../components/Center.jsx';

describe('Center Component', () => {
  const props = {
    centerDetails: {
      id: "1",
      name: "",
      description: {
        substr: () => {}
      },
      location: {
        substr: () => {}
      },
      capacity: "",
      price: "",
      available: 'true'
    },
    show: () => {}
  };
  const initialState = {
    eventReducer: {
      userEvents:[]
    },
    centerReducer: {
      allCenters:[]
    }
  };
  const store = mockStore(initialState);

  it('it should render Center component', () => {
  const wrapper = shallow(<Provider store={store} ><Center {...props} /></Provider>);
    expect(wrapper).toMatchSnapshot();
  });

  it('it should render Center component', () => {
    props.centerDetails.available = 'false';
    const wrapper = mount(<Provider store={store}><Center {...props} /></Provider>);
    expect(wrapper.length).toEqual(1);
  });

  it('it should handle onClick event', () => {
    const wrapper = mount(<Provider store={store} ><Center {...props} /></Provider>);
    let viewCenterButton = wrapper.find('.btn-primary');
    let submitButton = wrapper.find('.btn-success');
    viewCenterButton.simulate('click');
    submitButton.simulate('click');
  });
});
