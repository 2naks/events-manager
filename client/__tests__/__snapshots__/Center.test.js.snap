// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Center Component it should render Center component 1`] = `
<Center
  centerDetails={
    Object {
      "available": "true",
      "capacity": "",
      "description": Object {
        "substr": [Function],
      },
      "id": "1",
      "location": Object {
        "substr": [Function],
      },
      "name": "",
      "price": "",
    }
  }
  show={[Function]}
/>
`;
